### PR TITLE
Add Quiz Feature with Firebase Integration, Randomized Questions, and Testing

### PR DESCRIPTION
**Pull request by ziyadbagh**
### **Pull Request Description**

This pull request introduces a complete quiz feature to the application. The quizzes are dynamically fetched from Firebase, ensuring randomization of questions and options, and a new quiz screen has been added for users to interact with. Additionally, robust testing has been implemented, including both unit tests and UI tests, to ensure the reliability of the feature.

---

#### **Key Features:**

1. **Quiz Model and Firebase Integration**:
   - Created a `QuizQuestion` model representing a single quiz question, containing:
     - The correct answer (`correctWord`).
     - A list of incorrect options (`confusers`).
     - A list of ASL sign images (`signs`) dynamically generated based on the correct word.
   - Integrated Firebase Firestore to store and fetch quiz questions dynamically.
   - Implemented randomization logic for the options (correct answer + confusers) to ensure variety in each quiz.

2. **Quiz Screen**:
   - Added a new `QuizScreen` accessible from the home screen:
     - Displays the ASL signs representing the quiz word in a horizontally scrollable row.
     - Provides randomized multiple-choice options for the user to select from.
     - Includes a `Submit` button to validate the user's answer and, upon success, dynamically load the next quiz.
     - Handles empty or loading states gracefully.

3. **Navigation Integration**:
   - Linked the `QuizScreen` to the home screen, allowing seamless navigation into the quiz experience.

4. **Testing**:
   - **Unit Tests**:
     - Cover key logic, including Firestore document parsing, randomization of options, and ViewModel methods for quiz progression.
     - Ensured handling of edge cases, such as missing fields or empty datasets.
   - **UI Tests**:
     - Verified the complete user flow of the quiz, including:
       - Rendering ASL signs and multiple-choice options.
       - Correct and incorrect answer handling.
       - Navigation between quizzes upon success.
   - Achieved **86% line coverage** on new code.

5. **Error Handling and Logging**:
   - Added logging to identify issues with Firestore data (e.g., missing or invalid fields).
   - Graceful handling of cases where no quizzes are available or a fetch operation fails.

### CODE REVIEW

**Review by NBCCF025 — APPROVED** [review]
## Strengths:  
1. **Well-Defined Model and Integration**:  
   - The `QuizQuestion` model is robust and aligns well with the requirements.  
   - Dynamic fetching from Firebase with randomization logic is thoughtfully implemented.  

2. **User Experience**:  
   - The `QuizScreen` design provides an engaging and seamless user experience, especially with horizontally scrollable ASL signs and randomized options.  

3. **Testing Coverage**:  
   - Unit tests and UI tests appear comprehensive, covering both functional and edge cases.  
   - Achieving 85% coverage is commendable and indicates strong code reliability.

## Suggestions for improvement:
1. **UI Consistency**:  
   - Confirm that the visual design and navigation align with the rest of the application for a consistent user experience. For example using function from `Utils.kt`.

2. **Having more options**:
   - We could have more words in the database for more possible quizzes.

3. **Future Extensions**:  
   - It might be great to consider adding analytics to track user performance on quizzes that could be shown on their statistics.

## Summary:
**Short note**:
- I forgot to push changes on my own branch wich resulted to commit unrelated to this work that I corrected.

This is an impressive and comprehensive implementation of the quiz feature.
I already took care of the merge conflict as it was revolving on change I made for the navigation.
I approve this PR.

As you haven't put any image of your work, there is one :

quizScreenPRreview

File: app/src/main/java/com/github/se/signify/MainActivity.kt [hunk 1]
```diff
@@ -116,6 +117,18 @@ fun SignifyAppPreview(
       }
     }
 
+    navigation(
+        startDestination = Screen.QUIZ,
```
[the remark this answers is not in the document]
Comment (ziyadbagh) [reply app/src/main/java/com/github/se/signify/MainActivity.kt]: Ok, Thank you!!
